packages/admin-next/dashboard/src/routes/orders/returns-create/components/create-returns/returns-form.tsx

```diff
@@ -0,0 +1,410 @@
+          ret[variant.id] = variant.inventory[0]?.location_levels
+        })
+
+      return ret
+    }
+
+    getInventoryMap().then((map) => {
+      setInventoryMap(map)
+    })
+  }, [items])
+
+  const selectedLocation = form.watch("location")
+
+  const showLevelsWarning = useMemo(() => {
+    if (!selectedLocation) {
+      return false
+    }
+
+    const allItemsHaveLocation = items
+      .map((item) => {
+        if (!item?.variant_id) {
+          return true
+        }
+        return inventoryMap[item.variant_id]?.find(
+          (l) => l.location_id === selectedLocation
+        )
+      })
+      .every(Boolean)
+
+    return !allItemsHaveLocation
+  }, [items, inventoryMap, selectedLocation])
+
+  const {
+    quantity,
+    shipping,
+    custom_shipping_price,
+    enable_custom_shipping_price,
+  } = useWatch()
+
+  const shippingPrice = useMemo(() => {
+    if (enable_custom_shipping_price && custom_shipping_price) {
+      return getDbAmount(custom_shipping_price, order.currency_code)
+    }
+
+    const method = shipping_options?.find(
+      (o) => form.watch("shipping") === o.id
+    )
+
+    return method?.price_incl_tax || 0
+  }, [shipping, custom_shipping_price, enable_custom_shipping_price])
+
+  const refundable = useMemo(() => {
+    const itemTotal = items.reduce((acc: number, curr: LineItem): number => {
+      const unitRefundable =
+        (curr.refundable || 0) / (curr.quantity - curr.returned_quantity)
+
+      return acc + unitRefundable * quantity[curr.id]
+    }, 0)
+
+    const amount = itemTotal - (shippingPrice || 0)
+    onRefundableAmountChange(amount)
+
+    return amount
+  }, [items, quantity, shippingPrice])
+
+  useEffect(() => {
+    form.setValue("enable_custom_shipping_price", false)
+    form.setValue("custom_shipping_price", 0)
+  }, [form.watch("enable_custom_refund")])
+
+  return (
+    <div className="flex size-full flex-col items-center overflow-auto p-16">
+      <div className="flex w-full max-w-[736px] flex-col justify-center px-2 pb-2">
+        <div className="flex flex-col gap-y-1 pb-10">
+          <Heading className="text-2xl">{t("general.details")}</Heading>
+        </div>
+        <Heading className="mb-2 text-base">
+          {t("orders.returns.chooseItems")}
+        </Heading>
+        {items.map((item) => (
+          <ReturnItem
+            key={item.id}
+            item={item}
+            form={form}
+            currencyCode={order.currency_code}
+          />
+        ))}
+
+        <div className="flex flex-col gap-y-1 pb-4 pt-8">
+          <Heading className="text-base">{t("fields.shipping")}</Heading>
+        </div>
+
+        <div className="flex gap-x-4">
+          <div className="flex-1">
+            <Text weight="plus" className="txt-small">
+              {t("fields.location")}
+            </Text>
+            <Text className="text-ui-fg-subtle txt-small mb-1">
+              {t("orders.returns.locationDescription")}
+            </Text>
```

Comment:
move these too <Form.Label> and <Form.Hint> so they work correctly for screen readers